src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -372,6 +442,37 @@ private static string CreateMediaTypeWithEncoding(StringSegment mediaType, Encod
             return $"{mediaType.Value}; charset={encoding.WebName}";
         }
 
+        private bool MatchesSubtypePattern(MediaType subtypePattern)
+        {
+            if (subtypePattern.MatchesAllSubTypes)
+            {
+                return true;
+            }
+
+            if (subtypePattern.SubTypeSuffix.HasValue)
+            {
+                if (SubTypeSuffix.HasValue)
+                {
+                    // Both the pattern and the media type being checked have suffixes, so both parts must match.
+                    var matchesSubtypeWithoutSuffix = subtypePattern.MatchesAllSubTypesWithoutSuffix ||
+                        subtypePattern.SubTypeWithoutSuffix.Equals(SubTypeWithoutSuffix, StringComparison.OrdinalIgnoreCase);
+                    var matchesSubtypeSuffix = subtypePattern.MatchesAllNonemptySubTypeSuffixes ||
```

Comment:
Don't we need to check that `this` has a non-empty subtype suffix along with this check?